lib/puppet/functions/seeded_rand_string.rb

```diff
@@ -0,0 +1,29 @@
+# Generates a consistent random string of specific length based on (optionally) provided seed.
+#
+# @example Generate a consistently random string of length 8 with a seed:
+#   seeded_rand_string(8, "${module_name}::redis_password")
+#
+# @example Generate a random string from a specific set of characters:
+#   seeded_rand_string(5, '', 'abcdef')
+Puppet::Functions.create_function(:seeded_rand_string) do
+  # @param length Length of string to be generated.
+  # @param seed Seed string.
+  # @param charset String that contains characters to use for the random string.
+  #
+  # @return [String] Random string.
+  dispatch :rand_string do
+    param 'Integer[1]', :length
+    param 'String', :seed
```

Comment:
The text description states that seed is optional, but this declaration is for a required param.